### PR TITLE
Wi 50 collection cindy

### PR DESCRIPTION
**Pull request by Cindy088**
pull request to dev

### CODE REVIEW

File: src/collect/collect.service.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    const newCollect = new this.collectModel(userCollectDto);
+    return this.collectModel.create(newCollect);
+  }
+
+  //delete a collect
+  async deleteCollect(userCollectDto: UserCollectDto) {
+    return await this.collectModel.deleteOne({ userCollectDto });
+  }
+
+  // async checkCollect(CollectDTO: CollectDTO) {
```
Comment (Chanran) [inline src/collect/collect.service.ts:38]: delete useless code

File: src/collect/collect.service.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  async checkCollect(userCollectDto: UserCollectDto) {
+    const checkCollectData = await this.collectModel.findOne(userCollectDto);
+    if (checkCollectData) return this.deleteCollect(userCollectDto);
```
Comment (Chanran) [inline src/collect/collect.service.ts:41]: you can use if else instead of two if

File: src/collect/schemas/collect.schema.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+export type CollectDocument = HydratedDocument<Collect>;
+
+@Schema({ timestamps: true, versionKey: false })
+export class Collect {
+  _id: mongoose.Schema.Types.ObjectId;
+
+  //User A
+  @Prop()
+  collect_user_email: string; //from guard
```
Comment (Chanran) [inline src/collect/schemas/collect.schema.ts:12]: What's the difference between collect_user_email and collected_user_email?

Comment (Cindy088) [reply src/collect/schemas/collect.schema.ts:12]: collect_user (user A)  will collect/cancel collect collected_user's (user B) post.
and user email is the unique identified user id in our project.

**Comment by harryzuiku** [comment]
code review passed

**Review by harryzuiku — APPROVED** [review]

**Pull request merged**